devito/arch/compiler.py

```diff
@@ -737,6 +737,54 @@ def __init__(self, *args, **kwargs):
             warning("Running on Intel KNL without OpenMP is highly discouraged")
 
 
+class OneapiCompiler(IntelCompiler):
+
+    def __init__(self, *args, **kwargs):
+        super(OneapiCompiler, self).__init__(*args, **kwargs)
+
+        language = kwargs.pop('language', configuration['language'])
+        platform = kwargs.pop('platform', configuration['platform'])
+
+        self.cflags.append("-qopt-zmm-usage=high")
+
+        if language == 'sycl':
+            self.cflags.append('-fsycl')
+            if platform is NVIDIAX:
+                self.cflags.append('-fsycl-targets=nvptx64-cuda')
+            else:
+                self.cflags.append('-fsycl-targets=spir64')
+
+        if language == 'openmp':
+            # TODO: To be switched to `-fiopenmp` or `-qopenmp` as soon as
+            # it is fixed in the new Intel OneAPI release (current: 2023.0)
+            self.cflags.append('-fopenmp')
+            if platform is NVIDIAX:
+                self.cflags.append('-fopenmp-targets=nvptx64-cuda')
+            if platform is INTELGPUX:
+                self.cflags.append('-fopenmp-targets=spir64')
+                self.cflags.append('-fopenmp-target-simd')
+
+        if platform is INTELGPUX:
+            self.cflags.remove('-g')  # -g disables some optimizations in IGC
+            self.cflags.append('-gline-tables-only')
+            self.cflags.append('-fdebug-info-for-profiling')
+
+        # Make sure the MPI compiler uses `icx` underneath -- whatever the MPI distro is
```

Comment:
Get his through `super`

Reply:
indeed, otherwise what's the point of inhereting

Reply:
ping

also, rewrite warning msg to drop the "DPC++" thing